ygen/protogen.go

```diff
@@ -337,17 +463,18 @@ func writeProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *g
 // package for the protobuf message(s) that are being generated, such that relative
 // paths can be used in the messages.
 // TODO(robjs): Split the logic of this function into multiple subfunctions.
```

Comment:
The situation with this function is becoming dire - it's really hard to keep track of what's going on due to the length. Consider creating an issue if you don't want to address it in this PR.

Reply:
Yeah -  this wasn't cool. I've gone and fixed this, refactored into specific functions. They have some inplace modification - but I think it is clear what's happening. PTAL and LMK if you don't think so. This refactor also changed a bunch of structs that were being passed to ptrs to those structs, and made much better reuse of the config struct, rather than recreating it all over the place.

Reply:
Changes look good, this is greatly improved :+1: